sentry_streams/src/messages.rs

```diff
@@ -0,0 +1,434 @@
+    let seq = list.into_sequence();
+    Ok(seq.unbind())
+}
+
+/// Represents a message with any Python object as payload. This message type
+/// is meant to be processed by Python operators. Rust operators should consider
+/// The payload as opaque and should not transform it.
+///
+/// The message can be created both by Python code or Rust code. When Rust code
+/// instantiates it, it has to convert it into a Py<PyAnyMessage> before handing
+/// it to Python code. This can be done with the `into_pyany` function.
+#[pyclass]
+#[derive(Debug)]
+pub struct PyAnyMessage {
+    #[pyo3(get)]
+    pub payload: Py<PyAny>,
+
+    pub headers: Vec<(String, Vec<u8>)>,
+
+    #[pyo3(get)]
+    pub timestamp: f64,
+
+    #[pyo3(get)]
+    pub schema: Option<String>,
+}
+
+pub fn into_pyany(py: Python<'_>, message: PyAnyMessage) -> PyResult<Py<PyAnyMessage>> {
+    // Converts a PyAnyMessage instantiated outside of Python memory into a
+    // Py<PyAnyMessage> that can be used in Python code.
+    let py_obj = Py::new(py, message)?;
+    Ok(py_obj)
+}
+
+#[pymethods]
+impl PyAnyMessage {
+    #[new]
+    pub fn new(
+        payload: Py<PyAny>,
+        headers: Py<PySequence>,
+        timestamp: f64,
+        schema: Option<String>,
+        py: Python<'_>,
+    ) -> PyResult<Self> {
+        Ok(Self {
+            payload,
+            headers: headers_to_vec(py, headers)?,
+            timestamp,
+            schema,
+        })
+    }
+
+    #[getter]
+    fn headers(&self, py: Python<'_>) -> PyResult<Py<PySequence>> {
+        headers_to_sequence(py, &self.headers)
+    }
+
+    fn replace_payload(&self, new_payload: Py<PyAny>) -> PyAnyMessage {
+        PyAnyMessage {
+            payload: new_payload,
+            headers: self.headers.clone(),
+            timestamp: self.timestamp,
+            schema: self.schema.clone(),
+        }
+    }
+
+    fn __repr__(&self, py: Python<'_>) -> PyResult<String> {
+        let payload_repr = self.payload.call_method0(py, "__repr__")?;
+        Ok(format!(
+            "PyAnyMessage(payload={}, headers={:?}, timestamp={}, schema={:?})",
+            payload_repr, self.headers, self.timestamp, self.schema
+        ))
+    }
+
+    fn __str__(&self, py: Python<'_>) -> PyResult<String> {
+        self.__repr__(py)
+    }
+}
+
+/// Represent a message whose payload is a byte array. The payload is a Vec<u8>, not
+/// a PyBytes. Copy is needed to convert one to the other. This is meant primarily to
+/// represent the message produced by a Rust source and consumed by a Rust Sink.
+///
+/// TODO: With FFI there should be a way to share a byte array between Rust and Python
+///       without copying.
+#[pyclass]
+#[derive(Debug)]
+pub struct RawMessage {
+    pub payload: Vec<u8>,
+
+    pub headers: Vec<(String, Vec<u8>)>,
+
+    #[pyo3(get, set)]
+    pub timestamp: f64,
+
+    #[pyo3(get, set)]
+    pub schema: Option<String>,
+}
+
+#[pymethods]
+impl RawMessage {
```

Comment:
All of the functions here seem to be for creating the RawMessage from Python objects, and converting the message into Python objects. Can we use this interface to go from Rust to Rust?

I might be misunderstanding how the Rust API handles messages.

Reply:
What I mean is, all the functions return Python object (the `headers` call for example). If this is always converted to a Python object, how does the Rust to Rust functionality work?

Reply:
Yes OK that makes sense.